### DISCUSSION TITLE
Access engine's let and const global variables

### DISCUSSION THREAD

**Discussion by huantianad** [sebastienros/jint#1194] [Q&A]
Is there a way to get variables declared with `let` and `const` in the global namespace? Example:

```cs
using Jint;

var engine = new Engine();

var baseVariableNames = engine.Realm.GlobalObject.GetOwnPropertyKeys();

engine.Execute("var a = 1");
engine.Execute("let b = 2");
engine.Execute("const c = 3");

var variableNames = engine.Realm.GlobalObject.GetOwnPropertyKeys();

Console.WriteLine(string.Join(
    "\n",
    variableNames
        .Where(varName => !baseVariableNames.Contains(varName))
        .Select(rawName => rawName.AsString())
        .Select(name => $"{name} = {engine.GetValue(name)}")
));
// Prints:
// a = 1

// Would like:
// a = 1
// b = 2
// c = 3 (maybe a note that it's const??)

```

**Comment by lahma** [comment]
There's no way to get lexically scoped variables as they are transient in nature (scoped to lexical context). You probably need to analyze scripts with `AstVisitor` or such. There's also [a debugger](https://github.com/Jither/Jint.DebugAdapter) that can tell what can be seen on the current stack frame.

**Reply by huantianad** [reply]
That's what I was afraid of. I don't know if I'll be able to do static analysis on the script itself. I'll look into the debugger and see what's going on over there tho!
